src/avram/base_query_template.cr

```diff
@@ -85,20 +85,30 @@ class Avram::BaseQueryTemplate
         {% end %}
 
 
-        def where_{{ assoc[:table_name] }}
-          {{ assoc[:type] }}::BaseQuery.new_with_existing_query(query).tap do |assoc_query|
-            yield assoc_query
-          end
+        def where_{{ assoc[:table_name] }}(assoc_query : ::{{ assoc[:type] }}::BaseQuery, join auto_join : Bool = true)
+          join_{{ assoc[:table_name] }} if auto_join
```

Comment:
@edwardloveall Here is the code we're talking about. Previously you had to do the join yourself or the where would not work: `some_query.join_comments.where_comments(..)`. This makes it so that it calls `join_comments` for you so you don't need to do it every time. The problem is that sometimes you want a different kind of  join (not  an inner join). So in those cases you want to skip  the method automatically adding a join or it will be  the  equivalent of this:  `some_query.join_comments.left_join_comments.where_comments(...)`

Reply:
We could revert this behavior and have you explicitly join each time, but I don't like that very much because 99% of the time you want a regular inner join, and you won't get a compile time error if you forget. Laravel does the join for you when querying associations which I think is pretty cool